Uploader.m

```
@@ -0,0 +1,152 @@
+    [req setValue:contentType forHTTPHeaderField:@"Content-Type"];
+    for (NSString *key in _params.headers) {
+        id val = [_params.headers objectForKey:key];
+        if ([val respondsToSelector:@selector(stringValue)]) {
+            val = [val stringValue];
+        }
+        if (![val isKindOfClass:[NSString class]]) {
+            continue;
+        }
+        [req setValue:val forHTTPHeaderField:key];
+    }
+
+    NSData *formBoundaryData = [[NSString stringWithFormat:@"--%@\r\n", formBoundaryString] dataUsingEncoding:NSUTF8StringEncoding];
+    NSMutableData* reqBody = [NSMutableData data];
+
+    // add fields
+    for (NSString *key in _params.fields) {
+        id val = [_params.fields objectForKey:key];
+        if ([val respondsToSelector:@selector(stringValue)]) {
+            val = [val stringValue];
+        }
+        if (![val isKindOfClass:[NSString class]]) {
+            continue;
+        }
+
+        [reqBody appendData:formBoundaryData];
+        [reqBody appendData:[[NSString stringWithFormat:@"Content-Disposition: form-data; name=\"%@\"\r\n\r\n", key] dataUsingEncoding:NSUTF8StringEncoding]];
+        [reqBody appendData:[val dataUsingEncoding:NSUTF8StringEncoding]];
+        [reqBody appendData:[@"\r\n" dataUsingEncoding:NSUTF8StringEncoding]];
+    }
+
+    // add files
+    for (NSDictionary *file in _params.files) {
+        NSString *name = file[@"name"];
+        NSString *filename = file[@"filename"];
+        NSString *filepath = file[@"filepath"];
+        NSString *filetype = file[@"filetype"];
+
+        // Check if file exists
+        NSFileManager *fileManager = [NSFileManager defaultManager];
+        if (![fileManager fileExistsAtPath:filepath]){
+            // NSError* error = [NSError errorWithDomain:@"Uploader" code:NSURLErrorFileDoesNotExist userInfo:@{NSLocalizedDescriptionKey: [NSString stringWithFormat: @"Failed to open target file at path: %@", filepath]}];
```

Any reason why you are swallowing this error here and just continue? I feel the user should get feedback in JS-land if an upload fails due to the file not existing.